src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -403,11 +398,28 @@ void Replayer<I>::handle_load_local_group_snapshots(int r) {
       continue;
     }
     if (ns->state != cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY) {
+      if (ns->is_orphan()) {
+        dout(5) << "local group being force promoted" << dendl;
+        locker.unlock();
+        handle_replay_complete(0, "orphan (force promoting)");
+        return;
+      } else if (m_retry_validate_snap) {
+        m_retry_validate_snap = false;
+        locker.unlock();
+        schedule_load_group_snapshots();
+        return; // previous snap is syncing, reload local snaps to confirm its completeness.
+      }
       break;
+    } else  { // primary
+      if (it->state == cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE) {
+        dout(10) << "found incomplete primary group snapshot: "
+                 << it->id << dendl;
+        continue;
```

Comment:
Shouldn't this break and reload the local snaps ?

Reply:
No. This is as per what we discussed to do in:

```
 librbd: group get info skip incomplete snapshot

group get info shouldn't look at incomplete group snapshot if group is
primary or primary demoted. For example a failed force promote can leave an
incomplete primary group snapshot on the group, it doesn't mean that group
is primary yet.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>
```

Reply:
This is not group get info, this is the Replayer. The GR should reload the snaps until the primary state moves to complete.